modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/EditFolderMVCActionCommand.java

```diff
@@ -245,6 +259,74 @@ private void _updateFolder(ActionRequest actionRequest)
 		}
 	}
 
+	private void _updatePropagationPermission(
+			ActionRequest actionRequest, Folder folder,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		if (folder == null) {
+			return;
+		}
+
+		DLFolder dlParentFolder = _dlFolderLocalService.fetchDLFolder(
+			folder.getParentFolderId());
+
+		if ((dlParentFolder == null) ||
+			!dlParentFolder.isPermissionPropagationEnabled()) {
+
+			return;
+		}
+
+		Portlet portlet = _portletLocalService.getPortletById(
+			serviceContext.getCompanyId(), serviceContext.getPortletId());
+
+		PermissionPropagator permissionPropagator =
+			portlet.getPermissionPropagatorInstance();
+
+		if (permissionPropagator == null) {
+			return;
+		}
+
+		int[] roleTypes = RoleConstants.TYPES_REGULAR_AND_SITE;
+
+		Group group = _groupLocalService.fetchGroup(
+			dlParentFolder.getGroupId());
+
+		if ((group != null) && group.isDepot()) {
+			roleTypes = new int[] {
+				RoleConstants.TYPE_REGULAR, RoleConstants.TYPE_DEPOT
+			};
+		}
+
+		if (group != null) {
```

Comment:
If the group is asset library (depot), is it redundant when we check the role types whether it is organization.

Reply:
I will fix it